### PR TITLE
sole: add version 1.0.4 and support conan v2

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **sole/***

Retry https://github.com/conan-io/conan-center-index/pull/13484

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/sole/all/conanfile.py [hunk 1]
```diff
@@ -1,39 +1,54 @@
...
 
 class SoleConan(ConanFile):
     name = "sole"
-    homepage = "https://github.com/r-lyeh-archived/sole"
     description = "Sole is a lightweight C++11 library to generate universally unique identificators (UUID), both v1 and v4."
-    topics = ("conan", "uuid", "header-only")
+    license = "Zlib"
     url = "https://github.com/conan-io/conan-center-index"
+    homepage = "https://github.com/r-lyeh-archived/sole"
+    topics = ("conan", "uuid", "header-only")
```
Comment (prince-chrismc) [inline recipes/sole/all/conanfile.py:16]:
```suggestion
    topics = ("uuid", "header-only")
```

File: recipes/sole/all/conanfile.py [hunk 2]
```diff
@@ -1,39 +1,54 @@
...
     settings = "os", "compiler"
```
Comment (prince-chrismc) [inline recipes/sole/all/conanfile.py:17]:
```suggestion
    settings = "os", "arch", "compiler", "build_type"
```

**Review by prince-chrismc — COMMENTED** [review]
two tiny nits otherwise ✅

File: recipes/sole/all/conanfile.py (hunk 1 above)
Comment (toge) [reply recipes/sole/all/conanfile.py]: Fixed

File: recipes/sole/all/conanfile.py (hunk 2 above)
Comment (toge) [reply recipes/sole/all/conanfile.py]: Thank you for your comments.
Fixed.

**Review by jwillikers — APPROVED** [review]

**Review by prince-chrismc — APPROVED** [review]

**Pull request merged**